src/inbox.rs

```diff
@@ -70,6 +70,22 @@ impl<A> Chan<A> {
         self.shutdown_waiting_senders();
     }
 
+    pub fn increment_sender_count(&self) {
+        // Memory orderings copied from Arc::clone
+        self.sender_count.fetch_add(1, atomic::Ordering::Relaxed);
+    }
+
+    pub fn decrement_sender_count(&self) {
```

Comment:
I like this change overall, but this name I find to be a little misleading now - it has extra behaviour in some cases beyond just decrementing the refcount

Reply:
I am open to suggestions :)

We could name it after an event handler / hook, like `on_sender_dropped` and the other one `on_sender_created`?

Reply:
I think that makes sense!